[docs] EAS Build - update info about default images (#17451)
expo/expo: docs/pages/build-reference/infrastructure.md

> **Note:**
>

> > If you do not provide `image` in **eas.json** your build is going to use `default` image, but in some cases, we are selecting a more appropriate image based on SDK version or react-native version. You can check what image is used for a build in the "Spin up build environment" build logs section.

Reviewer: ```suggestion
> If you do not provide `image` in **eas.json**, your build will use `default` image. However, in some cases, we select a more appropriate image based on the Expo SDK version or React Native version. You can check what image is used for a build in the "Spin up build environment" build logs section.
```
